### PR TITLE
voting key link transaction change

### PR DESCRIPTION
**Pull request by gimer**
should SDKs be able to support old transactions?
I think not, but this means upgrade should be released as soon as server will be released.

### CODE REVIEW

**Comment by fboucquez** [comment]
If we do this change, the account_state may change its size also, PinnedVotingKey would be smaller and we won't be able to serialize/deserialize old states. I know a smaller voting key won't be an issue after public release, but do we need to add a version to the state schemas in general for future transactions/state changes? If we change a transaction that produces a different state schema, should the schema be treated differently also based on a version?

**Comment by gimer** [comment]
updated

File: schemas/coresystem/voting_key_link_v1.cats [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	const EntityType entityType = 0x4143
+
+	inline Transaction
+	inline VotingKeyLinkV1TransactionBody
+
+# binary layout for an embedded voting key link transaction
+struct EmbeddedVotingKeyLinkV1Transaction
+	inline EmbeddedTransaction
+	inline VotingKeyLinkV1TransactionBody
+
```
Comment (Jaguar0625) [inline schemas/coresystem/voting_key_link_v1.cats:29]: delete trailing line

**Review by Jaguar0625 — APPROVED** [review]
i assume you want to merge into `testnet` (not `main`) ?

File: schemas/coresystem/voting_key_link_v1.cats (hunk 1 above)
Comment (gimer) [reply schemas/coresystem/voting_key_link_v1.cats:29]: yup (it's rebased on testnet)

**Review by Jaguar0625 — APPROVED** [review]

**Comment by fboucquez** [comment]
import "coresystem/voting_key_link_v1.cats" 

is missing from all.cats

**Comment by fboucquez** [comment]
Is it possible to update this branch or the testnet branch with the latest fixes from main (root alias and versions in states)

**Comment by gimer** [comment]
rebased and merged into teatnet branch

**Pull request merged**